### PR TITLE
(fix) Allow read partitions with extra = in the value

### PR DESCRIPTION
**Pull request by kukushking**
### Feature or Bugfix
- Bugfix

### Detail
- #1778 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 65fbc9efed43863ab3f9e24eb1217c45da50f97b
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5e94023c-3029-4ce3-a8fa-f767a72e7d73%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]
👍🚀

File: awswrangler/s3/_read.py [hunk 1]
```diff
@@ -71,11 +71,11 @@ def _extract_partitions_from_path(path_root: str, path: str) -> Dict[str, str]:
     if path_root not in path:
         raise exceptions.InvalidArgumentValue(f"Object {path} is not under the root path ({path_root}).")
     path_wo_filename: str = path.rpartition("/")[0] + "/"
-    path_wo_prefix: str = path_wo_filename.replace(f"{path_root}/", "")
-    dirs: Tuple[str, ...] = tuple(x for x in path_wo_prefix.split("/") if (x != "") and (x.count("=") == 1))
+    path_wo_prefix: str = path_wo_filename.replace(f"{path_root}", "")
+    dirs: Tuple[str, ...] = tuple(x for x in path_wo_prefix.split("/") if x != "")
```
Comment (LeonLuttenberger) [inline awswrangler/s3/_read.py:75]: Don't we still want to check if `"=" in x`?

**Comment by kukushking** [comment]
Timestream tests again...

File: awswrangler/s3/_read.py (hunk 1 above)
Comment (kukushking) [reply awswrangler/s3/_read.py:75]: Oof, good catch

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 91974d432ae6edb7491bcbb0fe5f8b3c3f5e1923
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=12e5131c-6540-48c3-af55-1ea27642320f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**